casper/src/test/resources/PoSTest.rho

```diff
@@ -144,6 +146,54 @@ match (
               }
             }
           } |
+          contract test_withdraw_succeeds(rhoSpec, _, ackCh) = {
+            new setupCh, retCh,
+                initialPosBalanceCh, finalPosBalanceCh,
+                initialUserBalanceCh, finalUserBalanceCh,
+                initialRewardsCh, finalRewardsCh,
+                initialBondsCh, finalBondsCh,
+                withdrawCh, closeBlockCh in {
+              prepareUser!("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", 10000, *setupCh) |
+              for (@user1PubKey, _, @user1Vault <- setupCh) {
+                @posVault!("balance", *initialPosBalanceCh) |
+                @user1Vault!("balance", *initialUserBalanceCh) |
+                @PoS!("getBonds", *initialBondsCh) |
+                @PoS!("getRewards", *initialRewardsCh) |
+                for(@initialPosBalance <- initialPosBalanceCh;
+                    @initialUserBalance <- initialUserBalanceCh;
+                    @initialBonds <- initialBondsCh;
+                    @initialRewards <- initialRewardsCh) {
+                  @PoS!("bond", 100, *retCh) |
+                  for ( @(true, _) <- retCh) {
+                    @PoS!("withdraw", *withdrawCh) |
+                    for (_ <- withdrawCh) {
+                      @PoS!("closeBlock", *closeBlockCh) |
+                      for (_ <- closeBlockCh) {
+                        @PoS!("getBonds", *finalBondsCh) |
+                        @posVault!("balance", *finalPosBalanceCh) |
+                        @user1Vault!("balance", *finalUserBalanceCh) |
+                        @PoS!("getRewards", *finalRewardsCh) |
+                        for(@finalPosBalance <- finalPosBalanceCh;
+                            @finalUserBalance <- finalUserBalanceCh;
+                            @finalBonds <- finalBondsCh;
+                            @finalRewards <- finalRewardsCh) {
+                          rhoSpec!("assertMany",
+                            [
+                              ((finalBonds, "==", initialBonds), "bonds don't change"),
```

Comment:
Do these not change because the validator withdrawing has not yet completed the required quarantine length?

Reply:
Shouldn't there be a test that verifies these values _do_ change?

Reply:
There is no change because no payment happens between bond and withdraw. I'll add a test for a situation where things change